src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnection.cs

```diff
@@ -263,9 +263,17 @@ private void ConsumeFromRemainingBuffer(int bytesToConsume)
                         cookiesFromContainer = null;
                     }
 
+                    // Some headers such as User-Agent and Server use space as a separator (see: ProductInfoHeaderParser)
```

Comment:
Maybe wrap this and everything in this block below this point in:
```C#
if (header.Value.Length > 1)
{
    ...
}
```
It's minor, but there's no point in doing this work to get the right separator if we're not going to use it.